### PR TITLE
fix(ingest/json-schema): adding support descriptions for array

### PR DESCRIPTION
**Pull request by AvaniSiddhapuraAPT**
## Checklist

- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)  https://linear.app/acryl-data/issue/ING-492/json-schema-source-support-descriptions-for-array-and-map-types
- [ ] Tests for the changes have been added/updated (if applicable) New test has been added for the change. 
- [ ] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [ ] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

### CODE REVIEW

File: metadata-ingestion/src/datahub/ingestion/extractor/json_schema_util.py [hunk 1]
```diff
@@ -620,6 +658,29 @@ def get_fields_from_schema(
                 base_field_path=FieldPath(is_key_schema=is_key_schema),
             )
 
+    @classmethod
+    def extract_descriptions_from_schema(cls, schema: Dict[str, Union[str, Dict]]) -> List[Optional[str]]:
+        """
+        Extracts descriptions from a JSON schema.
+
+        Args:
+            schema (Dict[str, Union[str, Dict]]): The JSON schema.
```
Comment (hsheth2) [inline metadata-ingestion/src/datahub/ingestion/extractor/json_schema_util.py:667]: is this the same method as above?

Comment (AvaniSiddhapuraAPT) [reply metadata-ingestion/src/datahub/ingestion/extractor/json_schema_util.py:667]: Hey @hsheth2, Yeah. Oppsy. Let me fix things up here in this PR. Give me sometime please.

**Comment by AvaniSiddhapuraAPT** [comment]
@hsheth2 @anshbansal  I need a little guidance here, to figure out what is the error here. Also it would be great if you can provide 10-15 min touch base to understand the basic standard or expectation of testing on local before raising any PR, so that i can avoid such scenario's in future.

File: metadata-ingestion/tests/unit/schema/test_json_schema_util.py [hunk 2]
```diff
@@ -374,6 +379,8 @@ def test_nested_arrays():
     expected_field_paths: List[str] = [
         "[version=2.0].[type=NestedArray].[type=array].[type=array].[type=Foo].ar",
         "[version=2.0].[type=NestedArray].[type=array].[type=array].[type=Foo].ar.[type=integer].a",
+        "[version=2.0].[type=NestedArray].[type=array].[type=array].ar",
+        "[version=2.0].[type=NestedArray].[type=array].ar",
```
Comment (mayurinehate) [inline metadata-ingestion/tests/unit/schema/test_json_schema_util.py:383]: Curious, have you verified on Datahub UI whether descriptions on such complex fields are represented there correctly ?

For example, how does SchemaMetadata aspect generated for below json schema look like when ingested in DataHub ? Is the description for each level captured and represented correctly on UI?
```
{
  "type": "object",
  "title": "NestedArray",
  "namespace": "com.linkedin",
  "properties": {
    "ar": {
      "type": "array",
      "description": "This is description of array ar",
      "items": {
        "type": "array",
        "description": "This is description of inner array of ar",
        "items": {
          "type": "object",
          "title": "Foo",
          "description": "This is description of foo object",
          "properties": {
            "a": {
              "type": "integer",
              "description": "description of integer a"
            }
          }
        }
      }
    }
  }
}
```

Comment (sagar-salvi-apptware) [reply metadata-ingestion/tests/unit/schema/test_json_schema_util.py:383]: Hi @mayurinehate ,
Is this expected UI result you are looking for ?
Let me know if there any discrepancies  between UI result i shared and your expected result. 

abcScreenshot 2024-02-22 at 1 05 07 PM

Comment (AvaniSiddhapuraAPT) [reply metadata-ingestion/tests/unit/schema/test_json_schema_util.py:383]: Hi @mayurinehate 

Is this the expected UI result for the sample nested array you have shared? 
Please let me know if there is any discrepancies between your expected result and UI result i shared.  

abcScreenshot 2024-02-22 at 1 05 07 PM

Comment (mayurinehate) [reply metadata-ingestion/tests/unit/schema/test_json_schema_util.py:383]: This does not look right on UI. What was the original behavior before the changes in this PR ? I would image, it only showed the collapsible `ar[][]` and not the last two entries of `ar[][]` and `ar[]`. Is that right ?

Comment (AvaniSiddhapuraAPT) [reply metadata-ingestion/tests/unit/schema/test_json_schema_util.py:383]: Hey, My apologies.
yes, collapse is not available for other two entries. I believe you require the nested collapses for all these entries based on the input nested json. Please correct me if i misunderstood. 
Let me fix this up and ping you.

**Review by hsheth2 — APPROVED** [review]

**Pull request merged**